### DISCUSSION TITLE
Help deleting from supabase

### DISCUSSION THREAD

**Discussion by Ablasko32** [supabase/supabase#31326] [Questions]
i have a problem when deleting image from supabase storage.
so i have a bucket called "items".
when i delete item i wanted to also delete the image but i didnt work.

so i just put placeholders in function to be able to debug it.I can verify that the name of bucket is "items" path is in root of items bucket,so no folder structure inside.

i get empty array back [ ] and null for error.

i can upload to same bucket,i can download(with the same image path,just replace remove with download and drop array),only i cant delete.

this is function 

```javascript
 export async function deleteImage(bucket, imagePath) {
  // deletes image with provided bucket name,and image
  const { data, error } = await supabase.storage
    .from("items")
    .remove(["reddragon.jpeg"]);
  console.log("delete image data", data);
  return { data, error };
``` 

im loggin: delete image data []

i have RLS policies set like follows: (bucket_id = 'items'::text) for everything

i could use some help guys :)

**Comment by GaryAustin1** [comment]
How are you determining the file has not been deleted?
Delete does not return anything if the file is not found, if it does not meet RLS or when it deletes.  It is always empty array and no error.

You have to meet both delete and select policies (with Using clauses).

**Reply by Ablasko32** [reply]
I located the issue, it was "double" RLS policies .The issue was that my policies were written on storage not on storage.objects.
I added SELECT and DELETE on storage.objects and fixed the issue.

Thanks for your help!
